pkg/test/e2e/ipam/ipam_test.go

```diff
@@ -0,0 +1,284 @@
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/rand"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	datacenter = "kubermatic"
+	location   = "hetzner-hel1"
+	version    = utils.KubernetesVersion()
+	credential = "e2e-hetzner"
+)
+
+func TestIPAM(t *testing.T) {
+	ctx := context.Background()
+
+	seedClient, _, _, err := utils.GetClients()
+	if err != nil {
+		t.Fatalf("failed to get client for seed cluster: %v", err)
+	}
+
+	// login
+	masterToken, err := utils.RetrieveMasterToken(ctx)
+	if err != nil {
+		t.Fatalf("failed to get master token: %v", err)
+	}
+	masterClient := utils.NewTestClient(masterToken, t)
+
+	// create dummy project
+	t.Log("creating project...")
+	project, err := masterClient.CreateProject(rand.String(10))
+	if err != nil {
+		t.Fatalf("failed to create project: %v", err)
+	}
+	defer masterClient.CleanupProject(t, project.ID)
+
+	t.Log("creating first cluster...")
+	cluster1, err := createNewCluster(ctx, masterClient, seedClient, project.ID)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	t.Log("creating first IPAM Pool...")
+	ipamPool1, err := createNewIPAMPool(ctx, seedClient, "192.168.1.0/28", "range", 8)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	t.Log("checking IPAM Pool 1 allocation on first cluster...")
+	if !checkIPAMAllocation(t, ctx, seedClient, cluster1, ipamPool1.Name, kubermaticv1.IPAMAllocationSpec{
+		Type:      "range",
+		DC:        location,
+		Addresses: []string{"192.168.1.0-192.168.1.7"},
+	}) {
+		t.Fatalf("IPAM Allocation 1 wasn't created on cluster 1")
+	}
+
+	t.Log("creating second cluster...")
+	cluster2, err := createNewCluster(ctx, masterClient, seedClient, project.ID)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	t.Log("checking IPAM Pool 1 allocation on second cluster...")
+	if !checkIPAMAllocation(t, ctx, seedClient, cluster2, ipamPool1.Name, kubermaticv1.IPAMAllocationSpec{
+		Type:      "range",
+		DC:        location,
+		Addresses: []string{"192.168.1.8-192.168.1.15"},
+	}) {
+		t.Fatalf("IPAM Allocation 1 wasn't created on cluster 2")
+	}
+
+	t.Log("creating second IPAM Pool...")
+	ipamPool2, err := createNewIPAMPool(ctx, seedClient, "192.168.1.0/27", "prefix", 28)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	t.Log("checking IPAM Pool 2 allocation on first cluster...")
+	if !checkIPAMAllocation(t, ctx, seedClient, cluster1, ipamPool2.Name, kubermaticv1.IPAMAllocationSpec{
+		Type: "prefix",
+		DC:   location,
+		CIDR: "192.168.1.0/28",
+	}) {
+		t.Fatalf("IPAM Allocation 2 wasn't created on cluster 1")
+	}
+
+	t.Log("checking IPAM Pool 2 allocation on second cluster...")
+	if !checkIPAMAllocation(t, ctx, seedClient, cluster2, ipamPool2.Name, kubermaticv1.IPAMAllocationSpec{
+		Type: "prefix",
+		DC:   location,
+		CIDR: "192.168.1.16/28",
+	}) {
+		t.Fatalf("IPAM Allocation 2 wasn't created on cluster 2")
```

Comment:
this can be problematic - I am not sure if you can rely on this particular order:
 - 1st cluster `"192.168.1.0/28"`
 - 2nd cluster `"192.168.1.16/28"`
 It could also be the other way around.